src/actions/plugins/cleverF.ts

```diff
@@ -0,0 +1,253 @@
+    vimState.lastCommaRepeatableMovement = new MoveCleverFFindBackward(this.keysPressed, true);
+
+    if (!result) {
+      return failedMovement(vimState);
+    }
+    // Mark a position to repeat this command
+    posF = [result.line, result.character];
+    return result;
+  }
+}
+
+/**
+ *  When change cleverFBackwardMode by "F" command, this class is called
+ */
+@RegisterAction
+class MoveCleverFFindBackward extends BaseMovement {
+  override modes = [Mode.CleverFBackwardMode];
+  keys = ['<character>'];
+  public override async execActionWithCount(
+    position: vscode.Position,
+    vimState: VimState,
+    count: number
+  ): Promise<vscode.Position | IMovement> {
+    // Reset Mode
+    await vimState.setCurrentMode(previousMode);
+    if (previousMode === Mode.Visual && !arrayEqual(posF, [position.line, position.character])) {
+      if (vimState.lastVisualSelection) {
+        vimState.cleverF.startVisualPosition = [
+          vimState.lastVisualSelection?.start?.line,
+          vimState.lastVisualSelection?.start?.character,
+        ];
+      } else {
+        vimState.cleverF.startVisualPosition = [position.line, position.character];
+      }
+      vimState.cleverF.isStartVisualBackward = true;
+    }
+
+    const cleverF = new CleverF();
+    cleverF.updateDecorations(position, vimState.editor, this.keysPressed[0]);
+
+    count ||= 1;
+    const toFind = Notation.ToControlCharacter(this.keysPressed[0]);
+    const result = findHelper(vimState, position, toFind, count, 'backward');
+
+    // To repeat this command
+    vimState.lastSemicolonRepeatableMovement = new MoveCleverFFindBackward(this.keysPressed, true);
+    vimState.lastCommaRepeatableMovement = new MoveCleverFFindForward(this.keysPressed, true);
+
+    if (!result) {
+      return failedMovement(vimState);
+    }
+    // Mark a position to repeat this command
+    posF = [result.line, result.character];
+
+    return result;
+  }
+}
+
+export interface ICleverF {
+  updateDecorations(position: Position, editor: vscode.TextEditor, character: string): void;
+  clearDecorations(editor: vscode.TextEditor): void;
+  startVisualPosition: number[];
+  isStartVisualForward: boolean;
+  isStartVisualBackward: boolean;
+}
+
+export class CleverF implements ICleverF {
+  private decorations: vscode.DecorationOptions[] = [];
+  public startVisualPosition: number[] = [-1, -1];
+  public isStartVisualForward: boolean = false;
+  public isStartVisualBackward: boolean = false;
+  private static readonly decorationType = vscode.window.createTextEditorDecorationType({
+    color: 'red',
+  });
+
+  constructor() {
+    this.decorations = [];
+  }
+  public updateDecorations(position: Position, editor: vscode.TextEditor, character: string) {
+    this.decorations = [];
+    const sourceCode = editor.document.getText();
+    if (character === '') {
+      return;
+    }
+    const regex = '(' + character + ')';
```

Comment:
Need to escape special characters

Reply:
thank you for using this PR plugin!!!
understood, I will try to fix it.

Reply:
I fixed it.
How about it ?

Reply:
Seems to be working well. Thank you!